src/algorithms/digi/PulseNoise.h

```diff
@@ -41,8 +41,8 @@ class PulseNoise : public PulseNoiseAlgorithm, public WithPodConfig<PulseNoiseCo
   void process(const Input&, const Output&);
 
 private:
-  std::default_random_engine generator; // TODO: need something more appropriate here
-  dd4hep::detail::FalphaNoise m_noise;
+  mutable std::default_random_engine m_generator; // TODO: need something more appropriate here
+  mutable dd4hep::detail::FalphaNoise m_noise;    // FalphaNoise::operator() is not const
```

Comment:
This might (more) belong in the other PR (https://github.com/eic/EICrecon/pull/1833)
Is calling something mutable a non-ideal way of getting around the const ness of a function?

Reply:
Correct. That's why that one depends on this PR. The context was overlapping and I didn't think it was that necessary to split them and deal with the conflicts.

Reply:
Amended and removed this from scope. Will add in #1833 instead.